main/lsp/lsp_helpers.cc

```diff
@@ -93,13 +93,42 @@ unique_ptr<Location> LSPLoop::loc2Location(const core::GlobalState &gs, core::Lo
     return make_unique<Location>(uri, loc2Range(gs, loc));
 }
 
+int cmpPositions(const Position &a, const Position &b) {
+    const int line = a.line - b.line;
+    if (line != 0) {
+        return line;
+    }
+    return a.character - b.character;
+}
+
+int cmpLocations(const Location &a, const Location &b) {
+    const int fileCmp = a.uri.compare(b.uri);
+    if (fileCmp != 0) {
+        return fileCmp;
+    }
+    const int startCmp = cmpPositions(*a.range->start, *b.range->start);
+    if (startCmp != 0) {
+        return startCmp;
+    }
+    return cmpPositions(*a.range->end, *b.range->end);
+}
+
 vector<unique_ptr<Location>>
 LSPLoop::extractLocations(const core::GlobalState &gs,
                           const vector<unique_ptr<core::lsp::QueryResponse>> &queryResponses,
                           vector<unique_ptr<Location>> locations) {
     for (auto &q : queryResponses) {
         addLocIfExists(gs, locations, q->getLoc());
     }
+    // Dedupe locations
```

Comment:
We may report some identical matches from the AST and the CFG now, so we need to dedupe the list of locations returned for a "Find all references" request.